### PR TITLE
Export Sort and Pagination types.

### PR DESCRIPTION
**Pull request by LiranCohen**
Exporting Filter, Sort and Pagination types for consumption by external MessageStores.

Lower max maxDataSizeAllowedToBeEncoded. We are currently checking this against the dataSize, which is the binary data size. However when we store it we are base64 encoding the data, which has an approximate increase of 33%, this was causing some issues going right above the limit in SQL.

Upon some rough calculations of the 65k limit for most SQL - 33% reduction, I rounded down to the nice number of 42_000.

### CODE REVIEW

File: src/core/dwn-constant.ts [hunk 1]
```diff
@@ -3,5 +3,5 @@ export class DwnConstant {
    * The maximum size in bytes of raw data that will be returned as `encodedData`.
    * this is also the maximum size that we will store within MessageStore.
    */
-  public static readonly maxDataSizeAllowedToBeEncoded = 50_000;
+  public static readonly maxDataSizeAllowedToBeEncoded = 42_000;
```
Comment (diehuxx) [inline src/core/dwn-constant.ts:6]: the meaning of life 💫

Comment (diehuxx) [reply src/core/dwn-constant.ts:6]: I wonder if we should make this configurable in `Dwn.create()`. Made an issue so we can revisit the idea when we have time. #515

**Review by diehuxx — APPROVED** [review]

File: src/core/dwn-constant.ts (hunk 1 above)
Comment (thehenrytsai) [reply src/core/dwn-constant.ts:6]: Consider adding the context info above as comments directly in the code

Comment (LiranCohen) [reply src/core/dwn-constant.ts]: made the number more round with more headroom and added some comments to explain the size choice better.

**Review by thehenrytsai — APPROVED** [review]

**Pull request merged**